packages/nimble-components/src/wafer-map/specs/features/rendering.md

```diff
@@ -82,46 +82,50 @@ The POC is found in this branch [Worker Rendering POC](https://github.com/ni/nim
-    rowLengthsArray: Int32Array;
-    // the y coordinates of each die as a matrix row by row
-    dieRowIndexLayer: Int32Array;
-    // the value of each die as a matrix row by row
-    dieValuesLayer: Float64Array;
-    // the highlight  approach is still undecided, we have two options:
-    // the highlight state of each die as a matrix; user will have to pre-calculate tags into highlighted conditions.
-    dieHighlightsLayer: Int8Array;
-    // a 32 bitset array of tags for each die; aligns more closely to the existing public api but limits users to 32 tags.
-    dieHighlightsLayer: Int32Array;
-    // metadata array for each die; it will not be sent to the worker
-    metadata : unknown[]
+import { Table } from 'apache-arrow';
+export class WaferMap extends FoundationElement {
+...
+public diesTable: Table<{
+        colIndex: Int32,
+        rowIndex: Int32,
+        value: Float32,
+        tags: Uint32;
+        metadata: never;
+    }>
+...
 }
 ```
 
-Using TypedArrays has the benefit of direct transfer to web workers without structured cloning of the object by transferring the arrayBuffers and reconstructing the object. Other benefits of typedArrays include the low access time when iterating over the values, more memory efficiency and faster direct access to metadata layers values. The previous inputs can be adapted to this new structure to maintain backwards compatibility.
+This will be the [Apache Arrow](https://arrow.apache.org/docs/js/classes/Arrow_dom.Table.html) table schema.
+The row and column indices will be `Int32` columns, the values will be `Float32` columns.
+The tags for each die will be represented as a 32 bit mask stored in a `Uint32` column.
+The metadata column will be stored in an wildcard typed column.
 
-This API will have [optimized byte-array interop from Blazor](https://learn.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/6.0/byte-array-interop) and should be supported by Angular as a [vanilla javascript feature](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer).
+This approach has the benefits of a row based format that aligns well with the existing public API, as well as a nice public API that easily allows future improvements.
 
-The alternative to the above mentioned data structure is an [apache arrow](https://arrow.apache.org/docs/js/index.html) table with columns and metadata.
+The limits for this approach are the following:
 
-Pros of using Apache Arrow:
-
--   A row based format that aligns well with the existing public api
--   Well supported and tested format
--   Nice public API to use, we don't have to invent a new format, just document our schema for the arrow tables
--   Designed for large dataset visualizations
+1. There seems to be no support for columns of lists of strings. We decided to overcome this using a bit mask of tags. Another possible solution can be a dynamic number of columns for storing tags, but the performance may suffer.
```

Comment:
Could you go into more detail on how the bit mask of tags will impact clients? Will they be disappointed to lose the ability to specify arbitrary strings for tags? Or to specify more than 32 tags? Do we need to provide a library of functions to translate between an array of tag strings and a corresponding bit mask?

Reply:
After discussing with @rajsite and @atmgrifter00 we decided to try using dynamic columns for the table and highlights which can be joined and filtered using [arquero](https://uwdata.github.io/arquero/) in place of the bitmasks